### PR TITLE
Fix extractPurchaseDates crash

### PR DESCRIPTION
**Pull request by fire-at-will**
### Motivation
This PR fixes a crash in `CustomerInfo+ActiveDates.extractPurchaseDates()` caused by `Dictionary(uniqueKeysWithValues:)` when multiple purchases end up mapping to the same product key (for example with Google base plan IDs).

Instead of assuming uniqueness and crashing when the keys are not unique, we now merge duplicates and pick a preferred candidate deterministically.

This case should never occur during normal operations, but recently a minor backend bug caused this case to occur, and this PR updates the SDK to handle this failure scenario more gracefully.

### Description
This PR:
- Replaces `uniqueKeysWithValues` with `Dictionary(_:uniquingKeysWith:)` in `extractPurchaseDates()`.
- Added deterministic conflict resolution when mapped keys collide:

  1. Prefer non-nil purchase dates
  2. Prefer more recent purchase dates
  3. If tied, prefer Google-style raw product IDs (product:plan)
  4. If still tied, prefer entries with subscription metadata
  5. Final fallback is alphabetical order ascending by original product ID

### Tests
Added unit tests, including one that decodes CustomerInfo from a fixture with product IDs that cause the collision crash before this PR

### Next Steps
This PR does not remove all usages of `Dictionary(uniqueKeysWithValues:)` within the SDK. We should audit and remove other usages of `Dictionary(uniqueKeysWithValues:)` in future PRs.

---

> [!NOTE]
> **Low Risk**
> Behavior change is limited to edge-case merging of duplicate purchase keys and is covered by new unit tests; no auth/payment flow changes.
> 
> **Overview**
> Fixes a crash in `CustomerInfo.extractPurchaseDates` when multiple purchases map to the same product key (e.g., Google base plan IDs) by switching to `Dictionary(_:uniquingKeysWith:)` and deterministically selecting the preferred purchase-date candidate.
> 
> Adds unit coverage for Google-style mapping, duplicate-key resolution, non-subscription collisions, and a new fixture that previously triggered the crash, plus updates the Xcode project to include the new tests/fixture.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 8aa3ca01f188085b268f257b36ad77b2153e6ba0. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: Sources/Identity/CustomerInfo+ActiveDates.swift [hunk 1]
```diff
@@ -61,16 +61,62 @@ extension CustomerInfo {
...
+                    hasGoogleStyleProductID: productID.contains(":"),
+                    hasSubscriptionMetadata: purchase.expiresDate != nil || purchase.productPlanIdentifier != nil,
+                    originalProductID: productID
                 )
+                return (key, candidate)
+            },
+            uniquingKeysWith: preferredCandidate
+        )
+
+        return merged.mapValues(\.purchaseDate)
```
[the remark this answers is not in the document]
Comment (fire-at-will) [reply Sources/Identity/CustomerInfo+ActiveDates.swift:119]: FWIW, colons aren't allowed in iOS product IDs. This might be allowed on other platforms (stripe, roku, etc.) but not Apple!

Screenshot 2026-02-23 at 12 15 20 PM
